cli/flox/src/commands/init/python.rs

```diff
@@ -662,3 +671,236 @@ impl Display for ProvidedVersion {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use pretty_assertions::assert_eq;
+    use serial_test::serial;
+
+    use super::*;
+    use crate::commands::init::tests::FLOX_INSTANCE;
+
+    #[test]
+    fn test_should_run_true() {
+        let mut python = Python {
+            providers: vec![Provide::Found(Box::new(PoetryPyProject {
+                provided_python_version: ProvidedVersion::Compatible {
+                    requested: None,
+                    compatible: "3.11.6".to_string(),
+                },
+                poetry_version: "1.7.1".to_string(),
+            }))],
+            selected_provider: None,
+        };
+        assert!(python.should_run(Path::new("")).unwrap());
+    }
+
+    #[test]
+    fn test_should_run_false() {
+        let mut python = Python {
+            providers: vec![Provide::Invalid(anyhow!(""))],
+            selected_provider: None,
+        };
+        assert!(!python.should_run(Path::new("")).unwrap());
+    }
+
+    #[test]
+    fn test_get_init_customization_with_providers() {
+        let python = Python {
+            providers: vec![Provide::Found(Box::new(PoetryPyProject {
+                provided_python_version: ProvidedVersion::Compatible {
+                    requested: None,
+                    compatible: "3.11.6".to_string(),
+                },
+                poetry_version: "1.7.1".to_string(),
+            }))],
+            selected_provider: None,
+        };
+        assert_eq!(python.get_init_customization(), todo!());
+    }
+
+    #[test]
+    fn test_get_init_customization_with_selected_provider() {
+        let python = Python {
+            providers: vec![],
+            selected_provider: Some(Box::new(PoetryPyProject {
+                provided_python_version: ProvidedVersion::Compatible {
+                    requested: None,
+                    compatible: "3.11.6".to_string(),
+                },
+                poetry_version: "1.7.1".to_string(),
+            })),
+        };
+        assert_eq!(python.get_init_customization(), todo!());
+    }
```

Comment:
Could maybe drop these in favor of an e2e test

Reply:
wdym, shall we adapt the extend upon the `"'flox init' sets up a working Python environment using requirements.txt"` bats test instead?

I added bats tests in https://github.com/flox/flox/pull/1141 lmk if we also want to test verbatim output of these abstract hooks.